internal/verify/validate.go

```diff
@@ -20,6 +20,27 @@ var accountIDRegexp = regexp.MustCompile(`^(aws|aws-managed|\d{12})$`)
 var partitionRegexp = regexp.MustCompile(`^aws(-[a-z]+)*$`)
 var regionRegexp = regexp.MustCompile(`^[a-z]{2}(-[a-z]+)+-\d$`)
 
+func Valid4ByteASN(v interface{}, k string) (ws []string, errors []error) {
+	stringValue := strconv.Itoa(v.(int))
```

Comment:
@gdavison Can you see any concerns with the 32-bit BSD build on this?

Reply:
ASN numbers are 32-bit unsigned values, so this will fail for any values between `2147483647` and `4294967295`. Basically, for any resources using `terraform-plugin-sdk` (and therefore a `TypeInt`) we have to accept a `TypeString` and parse it into an `int64`. Technically, we could use a `uint32`, but the AWS API takes an `int64`